demos/classification_demo/main.cpp

```diff
@@ -0,0 +1,506 @@
+                accuracy = static_cast<double>(correctPredictionsCount) / framesNum;
+                gridMat.textUpdate(avgFPS, avgLatency, accuracy, isTestMode, !FLAGS_gt.empty(), presenter);
+                
+                if (!FLAGS_no_show) {
+                    cv::imshow("classification_demo", gridMat.outImg);
+                    key = static_cast<char>(cv::waitKey(1));
+                    presenter.handleKey(key);
+                }
+            } else if (!emptyInferRequests.empty()) {
+                lock.unlock();
+                auto inferRequestStartTime = std::chrono::steady_clock::now();
+                cv::Mat nextImage = resizeImage(inputImages[nextImageIndex], modelInputResolution);
+                emptyInferRequests.front().images.push_back(
+                                                    {nextImage,
+                                                     classIndices[nextImageIndex],
+                                                     inferRequestStartTime});
+                nextImageIndex++;
+                if (nextImageIndex == inputImagesCount) {
+                    nextImageIndex = 0;
+                }
+                if (emptyInferRequests.front().images.size() == FLAGS_b) {
+                    auto emptyInferRequest = emptyInferRequests.front();
+                    emptyInferRequests.pop();
+
+                    emptyInferRequest.inferRequest.SetCompletionCallback([emptyInferRequest,
+                                                      &completedInferRequests,
+                                                      &mutex,
+                                                      &condVar,
+                                                      &hasCallbackException,
+                                                      &irCallbackException] {
+                        try {
+                            std::lock_guard<std::mutex> callback_lock(mutex);
+                            completedInferRequests.push({emptyInferRequest.inferRequest, emptyInferRequest.images});
+                        }
+                        catch(...) {
+                            if (!hasCallbackException) {
+                                irCallbackException = std::current_exception();
+                                hasCallbackException = true;
+                            }
+                        }
+                        condVar.notify_one();
+                    });
+                    
+                    auto inputBlob = emptyInferRequest.inferRequest.GetBlob(inputBlobName);
+                    for (unsigned i = 0; i < FLAGS_b; i++) {
+                        matU8ToBlob<uint8_t>(emptyInferRequest.images[i].mat, inputBlob, i);
+                    }
+                    emptyInferRequest.inferRequest.StartAsync();
+                }
+            }
+            
+            lock.lock();
+            while (emptyInferRequests.empty() && completedInferRequests.empty()) {
```

Comment:
I think there's a possibility of a hang here. If all callbacks fail with an exception, then `completedInferRequests` will never become non-empty (and consequently, `emptyInferRequests` will never become non-empty either).